File: 0014-longest-common-prefix/solution.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        substring = strs[0]
        for word in strs:
            if len(word) < len(substring):
                word, substring = substring, word
            for i in range(len(substring)):
                if substring[i] != word[i]:
                    substring = substring[:i]
                    break
        return substring

File: 0014-longest-common-prefix/solution.c
char * longestCommonPrefix(char ** strs, int strsSize){
	int	letter = 0;
	char	common;
	while (strs[0][letter])
	{
		common = strs[0][letter];
		for (int word = 0; word < strsSize; word++)
		{
			if (common != strs[word][letter])
			{
				strs[0][letter] = '\0';
				return (strs[0]);
			}
		}
		letter++;
	}
	return (strs[0]);
}

